src/core/class/mediarithmics/common/ProductionPluginRunner.ts

```diff
@@ -1,27 +1,193 @@
 import { BasePlugin } from "./BasePlugin";
 import { Server } from "http";
+import * as cluster from "cluster";
+import { Credentials } from "../../../index";
+
+export enum MsgCmd {
+  CREDENTIAL_UPDATE_FROM_WORKER,
+  CREDENTIAL_UPDATE_FROM_MASTER,
+  LOG_LEVEL_UPDATE_FROM_WORKER,
+  LOG_LEVEL_UPDATE_FROM_MASTER,
+  GET_LOG_LEVEL_REQUEST
+}
+
+export interface SocketMsg {
+  cmd: MsgCmd;
+  value?: string;
+}
 
 export class ProductionPluginRunner {
+  numCPUs = require("os").cpus().length;
+
   pluginPort: number;
 
   plugin: BasePlugin;
   server: Server;
 
+  /**
+   * Socker Listener for master process. It has 4 jobs:
+   * 1 - If a new token is detected by a Worker, it receives a message and should send a message to each workers
+   * 2 - If a worker is asking for a token update (ex: the worker was just created because one of his friends died), the master should send a message to each workers
+   * 3 - If a log level change is detected by a worker, it receives a message and should send a message to each workers
+   * 4 - If a worker is asking for a log level update (ex: the worker was just created because one of his friends died), the master should send a message to each workers
+   * @param recMsg
+   */
+  masterListener = (worker: cluster.Worker, recMsg: SocketMsg) => {
+    this.plugin.logger.debug(
+      `Master ${process.pid} is being called with cmd: ${
+        MsgCmd[recMsg.cmd]
+      }, value: ${recMsg.value}`
+    );
+
+    // If we receive a Token Update, we update the token
+    if (recMsg.cmd === MsgCmd.CREDENTIAL_UPDATE_FROM_WORKER) {
```

Comment:
You can merge the same `ifs` and else if the credentials and logs

Reply:
You can also factorize the workers notifications and put the different processings depending on MsgCmd in separate methods.